File: leetcode/container-with-most-water.java
class Solution {
    fun maxArea(height: IntArray): Int {
        var left = 0
        var right = height.size - 1
        var maxArea = 0

        while (left < right) {
            val currentArea = minOf(height[left], height[right]) * (right - left)
            maxArea = maxOf(maxArea, currentArea)

            if (height[left] < height[right]) {
                left++
            } else {
                right--
            }
        }

        return maxArea
    }
}

File: leetcode/container-with-most-water.kt
class Solution {
    fun maxArea(height: IntArray): Int {
        var left = 0
        var right = height.size - 1
        var result = getExtend(left, right, height)
        while (left < right) {
            if (height[left] < height[right]) {
                left++
            } else if (height[right] < height[left]) {
                right--
            } else {
                left++
                right--
            }
            result = Math.max(result, getExtend(left, right, height))
        }
        return result
    }

    fun getExtend(left: Int, right: Int, height: IntArray): Int {
        return Math.min(height[left], height[right]) * (right - left)
    }
}